coderd/frobulators.go

```diff
@@ -0,0 +1,122 @@
+package coderd
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/httpapi"
+	"github.com/coder/coder/v2/coderd/httpmw"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// @Summary Get frobulators
+// @ID get-frobulators
+// @Security CoderSessionToken
+// @Param organization path string true "Organization ID"
+// @Param user path string true "User ID, name, or me"
+// @Produce json
+// @Tags Frobulator
+// @Success 200 {array} codersdk.Frobulator
+// @Router /organizations/{organization}/members/{user}/frobulators [get]
+func (api *API) listFrobulators(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	member := httpmw.OrganizationMemberParam(r)
+	org := httpmw.OrganizationParam(r)
+
+	frobs, err := api.Database.GetFrobulators(ctx, database.GetFrobulatorsParams{
+		UserID: member.UserID,
+		OrgID:  org.ID,
+	})
+	if err != nil {
+		httpapi.InternalServerError(rw, err)
+		return
+	}
+
+	out := make([]codersdk.Frobulator, 0, len(frobs))
+	for _, f := range frobs {
+		out = append(out, codersdk.Frobulator{
+			ID:          f.ID,
+			UserID:      f.UserID,
+			OrgID:       f.OrgID,
+			ModelNumber: f.ModelNumber,
+		})
+	}
+
+	httpapi.Write(r.Context(), rw, http.StatusOK, out)
+}
+
+// @Summary Post frobulator
+// @ID post-frobulator
+// @Security CoderSessionToken
+// @Param request body codersdk.InsertFrobulatorRequest true "Insert Frobulator request"
+// @Param organization path string true "Organization ID"
+// @Param user path string true "User ID, name, or me"
+// @Accept json
+// @Produce json
+// @Tags Frobulator
+// @Success 200 "New frobulator ID"
+// @Router /organizations/{organization}/members/{user}/frobulators [post]
+func (api *API) createFrobulator(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	member := httpmw.OrganizationMemberParam(r)
+	org := httpmw.OrganizationParam(r)
+
+	var req codersdk.InsertFrobulatorRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	frob, err := api.Database.InsertFrobulator(ctx, database.InsertFrobulatorParams{
+		UserID:      member.UserID,
+		OrgID:       org.ID,
+		ModelNumber: req.ModelNumber,
+	})
+	if err != nil {
+		httpapi.InternalServerError(rw, err)
+		return
+	}
```

Comment:
```suggestion
	if httpapi.Is404Error(err) { // Catches forbidden errors as well
		httpapi.ResourceNotFound(rw)
		return
	}
	if err != nil {
		httpapi.InternalServerError(rw, err)
		return
	}
```

Reply:
`httpapi.Is404Error(err)` catches forbidden errors. For security, we just return a 404 if you cannot access something, rather than a 401.

If we returned a 401 instead, you could determine the existence of something, which leaks information.